src/lib/UniversalDiscovery/Event/Subscriber/RichTextEmbedAllowedContentTypes.php

```diff
@@ -0,0 +1,100 @@
+
+class RichTextEmbedAllowedContentTypes implements EventSubscriberInterface
+{
+    /** @var array */
+    private $restrictedContentTypesIdentifiers;
+
+    /**
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \EzSystems\EzPlatformAdminUi\Permission\PermissionCheckerInterface $permissionChecker
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    public function __construct(
+        PermissionResolver $permissionResolver,
+        PermissionCheckerInterface $permissionChecker,
+        ContentTypeService $contentTypeService
+    ) {
+        $restrictedContentTypes = $this->getRestrictedContentTypesIdentifiers(
+            $permissionResolver,
+            $permissionChecker,
+            $contentTypeService
+        );
+
+        $this->restrictedContentTypesIdentifiers = count($restrictedContentTypes) ? $restrictedContentTypes : null;
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\PermissionResolver $permissionResolver
+     * @param \EzSystems\EzPlatformAdminUi\Permission\PermissionCheckerInterface $permissionChecker
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     */
+    private function getRestrictedContentTypesIdentifiers(
+        PermissionResolver $permissionResolver,
+        PermissionCheckerInterface $permissionChecker,
+        ContentTypeService $contentTypeService
+    ): array {
+        $access = $permissionResolver->hasAccess('content', 'read');
+        if (!\is_array($access)) {
+            return [];
+        }
+
+        $restrictedContentTypesIds = $permissionChecker->getRestrictions($access, ContentTypeLimitation::class);
+
+        if (empty($restrictedContentTypesIds)) {
+            return [];
+        }
+
+        $restrictedContentTypes = $contentTypeService->loadContentTypeList($restrictedContentTypesIds);
+
+        return array_values(array_map(function (ContentType $contentType): string {
+            return $contentType->identifier;
+        }, (array)$restrictedContentTypes));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ConfigResolveEvent::NAME => ['onUdwConfigResolve', -10],
+        ];
+    }
+
+    /**
+     * @param \EzSystems\EzPlatformAdminUi\UniversalDiscovery\Event\ConfigResolveEvent $event
+     */
+    public function onUdwConfigResolve(ConfigResolveEvent $event): void
+    {
+        $config = $event->getConfig();
+
+        if (!in_array($event->getConfigName(), ['richtext_embed', 'richtext_embed_image'])) {
+            return;
+        }
+
+        $config['allowed_content_types'] = $this->restrictedContentTypesIdentifiers;
```

Comment:
not following, why restricted content types are to be allowed?

Reply:
@alongosz sorry, have you tried to reproduce the scenario described in JIRA ticket?
Without this PR, if there are any restricted content types in `content`/`create` policy user will be able to select only those content types in UDW (please have a look at https://github.com/ezsystems/ezplatform-admin-ui/blob/2.0/src/lib/UniversalDiscovery/Event/Subscriber/ContentCreate.php).
In this PR we are checking if there are any restricted content types in `content`/`read` policy and we are using them instead of ones from `content`/`create`.

Reply:
I think it is about the naming, are those content types really restricted in `content/create`? Like from what I understand, your point is that user is restricted to create only those content types, right?
And from our (as you can see in config name - 'allowed_content_types` user is only allowed to create those).
It it only about naming the `restrictedContentTypesIdentifiers` property if I am right.

@SerheyDolgushev @alongosz @webhdx

Reply:
This is purely question of naming. If some Content Types are restricted to you, you definitely should not set them as allowed. If `restrictedContentTypesIdentifiers` contains identifiers from limitation, then I why don't you call it `allowedContentTypesIdentifiers`?

Reply:
Renamed in https://github.com/ezsystems/ezplatform-admin-ui/pull/1441/commits/d064c1a1013571b98f559231bab96499ddc88d35.

@ViniTou @alongosz Please note I was just trying to follow https://github.com/ezsystems/ezplatform-admin-ui/blob/d064c1a1013571b98f559231bab96499ddc88d35/src/lib/UniversalDiscovery/Event/Subscriber/ContentCreate.php#L23. And seems like it should be renamed as well. But it is out of scope for this PR.